src/marks/raster.js

```diff
@@ -340,34 +340,24 @@ function extrapolateBarycentric(W, I, X, Y, V, width, height, hull, index, mix)
   X = Float64Array.from(hull, (i) => X[index[i]]);
   Y = Float64Array.from(hull, (i) => Y[index[i]]);
   V = Array.from(hull, (i) => V[index[i]]);
-
-  const points = [];
-  const refs = [];
-  for (let j = 0; j < hull.length; ++j) {
-    const xa = X.at(j - 1);
-    const ya = Y.at(j - 1);
-    const dx = X.at(j) - xa;
-    const dy = Y.at(j) - ya;
-    const s = 1 / (1 + (Math.hypot(dx, dy) << 1));
-    for (let d = 0; d < 1; d += s) {
-      points.push(xa + d * dx, ya + d * dy);
-      refs.push(j);
-    }
-  }
-  const delaunay = new Delaunay(points);
-  let iy, ix;
+  const n = X.length;
+  const rays = Array.from({length: n}, (_, j) => ray(j, X, Y));
+  let k = 0;
```

Comment:
Maybe use the I array to note the value of `k` and retrieve it sparsely when starting a new line or emerging through the hull.

Reply:
I've tried this and it's not making a measurable difference in speed , so dropped it.